README.md

```diff
@@ -102,3 +102,128 @@ Waits for 1 second or given milliseconds
 await cli.wait()
 await cli.wait(3000)
 ```
+
+# cli.supertable
+
+Displays a customizable table
+
+```typescript
+cli.supertable.display(data, columns, options)
+```
+
+Where:
+
+- `data`: array of data objects to display
+- `columns`: Supertable.Columns object
+- `options`: Supertable.Options object
+
+`Supertable.Columns` defines the table columns and their display options. While the key is necessary, all column options are optional.
+
+```typescript
+const columns = Supertable.Columns = {
+  name: {},
+  id: {
+    header: 'ID', // override column header
+    minWidth: '10', // column must display at this width or greater
+    additional: true, // only display this column when the --additional flag is present
+    get: row => `US-O1-${row.id}`, // custom getter for data row object 
+  },
+}
+```
+
+`Supertable.Options` defines the table options, most of which are the parsed flags from the user for display customization, all of which are optional.
+
+```typescript
+const columns = Supertable.Options = {
+  printLine: customeLogger, // (optional) logger for csv displaying
+  columns: flags.columns,
+  sort: flags.sort,
+  filter: flags.filter,
+  csv: flags.csv,
+  additional: flags.additional,
+  'no-truncate': flags['no-truncate]',
+  'no-header': flags['no-header]',
+}
+```
+
+`cli.supertable.flags` is an object containing all the flags to include in your command class.
+
+```typescript
+{
+  columns: Flags.string({exclusive: ['additional'], description: 'only show provided columns (comma-seperated)'}),
+  sort: Flags.string({description: 'property to sort by (prepend \'-\' for descending)'}),
+  filter: Flags.string({description: 'filter property by partial string matching, ex: name=foo'}),
+  csv: Flags.boolean({exclusive: ['no-truncate'], description: 'output is csv format'}),
+  additional: Flags.boolean({description: 'show additional properties'}),
+  'no-truncate': Flags.boolean({exclusive: ['csv'], description: 'do not truncate output to fit screen'}),
+  'no-header': Flags.boolean({exclusive: ['csv'], description: 'hide table header from output'}),
+}
+```
+
+Example class:
+
+```typescript
+import {api} from 'my-api-clent'
+import {cli} from 'cli-ux'
+import {Command} from '@oclif/command'
+import {Supertable} from 'cli-ux/lib/styled/supertable'
+
+export default class Users extends Command {
+  static flags = {
+    ...cli.supertable.flags
+  }
+
+  async run() {
+    const {flags} = this.parse(User)
+    const users = await api.get<any[]>('/users')
+
+    const columns: Supertable.Columns = {
```

Comment:
I feel it would be cleaner to do:

```typescript
ux.supertable.display(apps, {
  // ...
}, {
  ...flags
})
```

My guess is that you wanted to be clear about how the types are defined, but I feel that's already clear above this example

Reply:
Hmm. Yeah, I wanted same interface as what was displayed under the `cli.supertable` header above to tie it all together.